scripts/build/types.js

```diff
@@ -61,5 +61,6 @@ if (!result.succeeded) {
       ` and ${result.warningCount} warnings`
   );
 
-  process.exitCode = 1;
+  // eslint-disable-next-line no-process-exit
+  process.exit(1);
```

Comment:
I don't think this is it. I've tried running this file and it does return code 1.
the issue comes from [here](https://github.com/algolia/instantsearch.js/blob/4ff0392767bfc7a2398b649b2f9362ddb9a8cc75/scripts/release/build-experimental-typescript.js#L23), where the code 1 returned by shell.exec is ignored.

Per [exec](https://www.npmjs.com/package/shelljs#examples) docs, 

```js
// Run external tool synchronously
if (shell.exec('git commit -am "Auto-commit"').code !== 0) {
  shell.echo('Error: Git commit failed');
  shell.exit(1);
}
```
this would effectively make sure that `scripts/release/build-experimental-typescript.js` return code 1 if any of the exec inside do.

Reply:
all right, no change is needed then, as the test is now ran separately too, so won't fail only on publish

Reply:
I did that change here: https://github.com/algolia/instantsearch.js/pull/4812